actors/evm/src/interpreter/precompiles.rs

```diff
@@ -22,13 +24,199 @@ lazy_static::lazy_static! {
     pub(crate) static ref SECP256K1: BigUint = BigUint::from_bytes_be(&hex_literal::hex!("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141"));
 }
 
+/// ensures top bits are zeroed
+fn assert_empty_bits<const S: usize>(src: &[u8]) -> Result<(), PrecompileError> {
+    if src[..S] != [0u8; S] {
+        Err(PrecompileError::InvalidInput)
+    } else {
+        Ok(())
+    }
+}
+
+struct Parameter<T>(pub T);
```

Comment:
A trait would fit better here.

Reply:
ideally yeah, but it helps a _lot_ to avoid conflicting implementations

Reply:
ah yeah that would work